userfiles/civet_virtual_study/civet_virtual_study.rb

```diff
@@ -27,56 +27,120 @@ class CivetVirtualStudy < CivetStudy
 
   Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
 
-  has_viewer :name => 'CIVET Virtual Study', :partial => :civet_virtual_study
+  CSV_BASENAME = "_civet_outputs.cbcsv"
 
-  # Sync the CivetVirtualStudy, with the CivetOutputs if deep=true
-  def sync_to_cache(deep=true)
+  reset_viewers # we don't need any of the superclass viewers
+  has_viewer :name => 'CIVET Virtual Study', :partial => :civet_virtual_study, :if => Proc.new { |u| u.is_locally_synced? }
+
+  # Sync the CivetVirtualStudy, with the CivetOutputs too if deep=true
+  def sync_to_cache(deep=true) #:nodoc:
+    return true if self.is_locally_synced?
     super()
     if deep
-      civet_outputs = get_civet_outputs
-      make_cache_symlinks(civet_outputs)
+      self.sync_civet_outputs
+      self.update_cache_symlinks
     end
+    @cbfl = @civet_outs = nil # flush internal cache
+    true
   end
 
-  # Load the list of CivetOutputs from the CbrainFileList
-  def get_civet_outputs
-    self.sync_to_cache(false)
+  # When syncing to the provider, we locally erase
+  # the symlinks, because they make no sense outside
+  # of the local Rails app.
+  # FIXME: this method has a slight race condition,
+  # after syncing to the provider we recreate the
+  # symlinks, but of another program tries to access
+  # them during that time they might not yet be there.
+  def sync_to_provider #:nodoc:
+    self.cache_writehandle do
+      self.erase_cache_symlinks
+    end
+    self.make_cache_symlinks
+    true
+  end
```

Comment:
Does this not have to call `super` at some point to perform the syncing to the provider?